extensions/ql-vscode/src/data-extensions-editor/extension-packs.ts

```diff
@@ -0,0 +1,99 @@
+import { relative } from "path";
+import { window } from "vscode";
+import { CodeQLCliServer } from "../cli";
+import { getOnDiskWorkspaceFolders, showAndLogErrorMessage } from "../helpers";
+import { ProgressCallback } from "../progress";
+
+export async function pickExtensionPack(
+  cliServer: CodeQLCliServer,
+  progress: ProgressCallback,
+): Promise<string | undefined> {
+  progress({
+    message: "Resolving extension packs...",
+    step: 1,
+    maxStep: 3,
```

Comment:
There's an implicit knowledge here that this function is called by something that has 3 steps (the `DataExtensionsEditorModule`). That feels a bit strange to me. Should we move the logic out of the module and into here?